### DISCUSSION TITLE
Using orama on the client

### DISCUSSION THREAD

**Discussion by fibonacid** [oramasearch/orama#528] [Q&A]
I have been asked to see if we can cut cost on search by moving it to the client.
I recognise that search is best done on a server, but since the content is not huge and we already plan to cache heavily on local storage i think it might be ok.

Since I have never used Orama i have a few questions:
- Can it run on the browser?
- Should i run it on a SW?
- Is there already a plugin that helps with this?
- Would a tanstack query adapter for Orama be a useful tool?

**Comment by stereobooster** [comment]
> Can it run on the browser?

Yes. https://github.com/orgs/oramasearch/discussions/186

> Should i run it on a SW?

Depends on what you want to do. I put it in [Web Worker](https://github.com/stereobooster/faceted-search/blob/main/src/pages/orama/oramaWorker.ts). Not sure though, why you would want to put it in Service Worker

> Is there already a plugin that helps with this?

With what specifically? There is a [plugin for Astro](https://docs.oramasearch.com/open-source/plugins/plugin-astro), for example, if you asking about using Orama as search for static website

> Would a tanstack query adapter for Orama be a useful tool?

Not really. You definitely can use it to load data inside Orama, but as well you can use anything else that you like (fetch, axios, etc.)

**Reply by fibonacid** [reply]
Thanks for the response! 

After experimenting a little bit with orama and react query I have found better ergonomics using a library called `minisearch`, which is designed for memory constraint environments.
